mountpoint-s3-fs/src/metrics.rs

```diff
@@ -90,15 +104,52 @@ fn poll_process_metrics(sys: &mut System) {
 #[derive(Debug)]
 struct MetricsSink {
     metrics: DashMap<Key, Metric>,
+    #[cfg(feature = "otlp_integration")]
+    otlp_exporter: Option<OtlpMetricsExporter>,
 }
 
 impl MetricsSink {
-    fn new() -> Self {
-        Self {
+    fn new(#[cfg(feature = "otlp_integration")] otlp_config: Option<OtlpConfig>) -> anyhow::Result<Self> {
+        #[cfg(feature = "otlp_integration")]
+        // Initialise the OTLP exporter if a config is provided
+        let otlp_exporter = if let Some(config) = otlp_config {
+            // Basic validation of the endpoint URL
+            if !config.endpoint.starts_with("http://") && !config.endpoint.starts_with("https://") {
+                return Err(anyhow::anyhow!(
+                    "Invalid OTLP endpoint configuration: endpoint must start with http:// or https://"
+                ));
+            }
+
+            #[cfg(feature = "otlp_integration")]
+            // Use TryInto to convert OtlpConfig to OtlpMetricsExporter
+            match std::convert::TryInto::<OtlpMetricsExporter>::try_into(&config) {
+                Ok(exporter) => {
+                    tracing::info!("OpenTelemetry metrics export enabled to {}", config.endpoint);
+                    Some(exporter)
+                }
+                Err(e) => {
+                    tracing::error!("Failed to initialise OTLP exporter: {}", e);
+
+                    // If the user explicitly requested metrics export but it failed,
+                    // we should return an error rather than silently continuing without metrics
+                    return Err(anyhow::anyhow!(
+                        "Failed to initialize OTLP metrics exporter: {}. If metrics export is not required, omit the OTLP configuration.",
+                        e
+                    ));
+                }
+            }
+        } else {
+            // No OTLP config provided, running without OpenTelemetry metrics export
+            tracing::debug!("Running without OpenTelemetry metrics export");
```

Comment:
Not blocking: Is this trace useful without the feature enabled? Might confuse people